Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Attachments/barrel_attachments.yml

```
@@ -109,6 +109,37 @@
         unwieldedOnly: true
       recoilFlat: -2
 
+- type: entity
+  parent: RMCBarrelAttachmentBase
+  id: RMCAttachmentShotgunChoke # choke
+  name: Shotgun Choke
+  description: "A threaded choke that can be inserted into the M42A2 Shotgun.
+    Improves accuracy, velocity, tightens spread, and increases the cyclic rate of most shotguns.
+    In exchange, projectile damage and force is reduced. Also increases recoil. Not recommended for slugs."
+  components:
+  - type: Sprite
+    state: choke
+  - type: Tag
+    tags:
+    - RMCAttachmentBarrel
+    - RMCAttachmentShotgunChoke
+  - type: AttachableVisuals
+  - type: GrantProjectileStunRemoveComponent
```

```suggestion
  - type: GrantProjectileStunRemove
```
The "component" part of the component's name should be left out in yaml